### PR TITLE
Rewrite SplitLocale to not allocate

### PR DESCRIPTION
**Pull request by Jacalz**
This improves performance and makes the code more readable. It also updates to use Go 1.19 as the base in `go.mod` so we get pruned module graphs (old Go compilers can still compile the code).

Benchstat:
```
goos: linux
goarch: amd64
pkg: github.com/fyne-io/go-locale
cpu: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz
              │   old.txt    │              new.txt               │
              │    sec/op    │   sec/op     vs base               │
SplitLocale-8   119.75n ± 1%   20.55n ± 1%  -82.84% (p=0.002 n=6)

              │  old.txt   │              new.txt              │
              │    B/op    │   B/op     vs base                │
SplitLocale-8   64.00 ± 0%   0.00 ± 0%  -100.00% (p=0.002 n=6)

              │  old.txt   │              new.txt               │
              │ allocs/op  │ allocs/op   vs base                │
SplitLocale-8   2.000 ± 0%   0.000 ± 0%  -100.00% (p=0.002 n=6)
```

### CODE REVIEW

**Comment by Jacalz** [comment]
Looks like the workflows med ro be updated to run on Go 1.19.

**Comment by Jacalz** [comment]
I can get that fixed later today.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/jeandeaual/go-locale
 
-go 1.14
+go 1.19
```
Comment (jeandeaual) [inline go.mod:3]:
```suggestion
go 1.18
```

**Comment by Jacalz** [comment]
I'm happy to help. You might get more contributions for the team given that we are planning on using your package in Fyne going forward :)

**Pull request merged**